actix-router/src/resource.rs

```diff
@@ -217,19 +452,19 @@ impl ResourceDef {
                     // path length === prefix segment length
                     path_len
                 } else {
-                    let is_slash_next =
-                        prefix.ends_with('/') || path.split_at(prefix.len()).1.starts_with('/');
-
-                    if path.starts_with(prefix) && is_slash_next {
+                    if path.starts_with(prefix)
+                        && (prefix.ends_with('/')
+                            || path.split_at(prefix.len()).1.starts_with('/'))
+                    {
                         // enters this branch if segment delimiter ("/") is present after prefix
                         //
                         // i.e., path starts with prefix segment
                         // and prefix segment ends with /
                         // or first character in path after prefix segment length is /
                         //
-                        // eg: Prefix("/test/") or Prefix("/test") would match:
-                        // - /test/foo
-                        // - /test/foo
+                        // eg: Prefix("/test/") or Prefix("/test") would match "/test/" and
+                        // "/test/foo" but Prefix("/test") would not process "/test" here since it
+                        // is handled by the earlier if case
 
                         if prefix.ends_with('/') {
                             prefix.len() - 1
```

Comment:
```suggestion
                        if prefix.ends_with('/') {
                            prefix.len()
```
I think this branch should also consume the full length of prefix. Otherwise we have inconsistent behavior with dynamic prefix `ResourceDef::prefix("/abc/").find_match("/abc/def") == Some(4)` vs `ResourceDef::prefix("/{id}/").find_match("/abc/def") == Some(5)`. I also think it is more intuitive behavior.

In this case this match arm would bre reduced to 
```rust
let rem = path.strip_prefix(prefix)?;

if prefix.ends_with('/') ||  rem.starts_with('/') || rem.is_empty() {
    Some(prefix.len())
} else {
    None
}
```